e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/effective-from-date.spec.js

```diff
@@ -0,0 +1,114 @@
+      cy.keyboardInput(bankRequestDatePage.bankRequestDateMonth(), dateConstants.todayMonth);
+      cy.keyboardInput(bankRequestDatePage.bankRequestDateYear(), dateConstants.todayYear);
+      cy.clickContinueButton();
+    });
+
+    it('should render page correctly', () => {
+      cy.url().should('eq', relative(`/case/${dealId}/cancellation/effective-from-date`));
+
+      cancelLink();
+      continueButton();
+      backLink();
+      effectiveFromDatePage.effectiveFromDateDay();
+      effectiveFromDatePage.effectiveFromDateMonth();
+      effectiveFromDatePage.effectiveFromDateYear();
+    });
+
+    it('should validate submitting a date more than 12 months in the future', () => {
+      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateDay(), dateConstants.twelveMonthsOneDayDay);
+      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateMonth(), dateConstants.twelveMonthsOneDayMonth);
+      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateYear(), dateConstants.twelveMonthsOneDayYear);
+
+      cy.clickContinueButton();
+      errorSummary().contains('The effective date cannot exceed 12 months in the future from the submission date');
+    });
+
+    it('should validate submitting a date more than 12 months in the past', () => {
+      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateDay(), dateConstants.twelveMonthsOneDayAgoDay);
+      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateMonth(), dateConstants.twelveMonthsOneDayAgoMonth);
+      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateYear(), dateConstants.twelveMonthsOneDayAgoYear);
+
+      cy.clickContinueButton();
+      errorSummary().contains('The effective date cannot exceed 12 months in the past from the submission date');
+    });
+
+    it('back link should take you to the bank request date page', () => {
+      cy.clickBackLink();
+
+      cy.url().should('eq', relative(`/case/${dealId}/cancellation/bank-request-date`));
+    });
+
+    it('continue button should take you to the check answers page', () => {
+      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateDay(), dateConstants.todayDay);
+      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateMonth(), dateConstants.todayMonth);
+      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateYear(), dateConstants.todayYear);
+
+      cy.clickContinueButton();
+
+      cy.url().should('eq', relative(`/case/${dealId}/cancellation/check-details`));
+    });
+
+    // TODO: DTFS2-7359 - add this test once cancel link is implemented
+    it.skip('cancel link should take you to confirm cancellation page', () => {
+      cy.clickCancelLink();
+
+      cy.url().should('eq', relative(`/case/${dealId}/cancellation/cancel`));
+    });
+  });
+
+  describe('when logged in as a non-PIM user', () => {
+    beforeEach(() => {
+      cy.login(T1_USER_1);
+    });
+
+    it('should redirect when visiting effective from date page ', () => {
```

Comment:
Minor, empty space at the end here 😛

Reply:
Good spot, will fix in next PR as suggested :)